.github/workflows/ci.yml

```diff
@@ -1,14 +1,23 @@
-name: Build
+name: Continuous Integration
 
 on:
   workflow_dispatch:
   push:
-    branches: [ master ]
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
 
 jobs:
-  build:
+
+  test:
     runs-on: ubuntu-latest
     steps:
+
+      - name: Setup SurrealDB
+        run: docker run -d --rm --network host surrealdb/surrealdb:latest start --user=root --pass=root
```

Comment:
You need to expose ports here, like in this repo
https://github.com/surrealdb/surrealdb.go/blob/main/.github/workflows/test.yml#L28

Reply:
Please notice the network flag, I used `host` network. But I changed it anyway to the default bridge network and mapped the port from the bridge to the host. This is to match other repos.